### PR TITLE
feat(react-instantsearch-v7): remove deprecated APIs

### PR DESCRIPTION
**Pull request by sarahdayan**
This removes the deprecated translation for the `<Stats>` widget.

I also removed the `warn` export from `react-instantsearch-hooks` since we don't want users to start using it (unlikely, but better avoid it if we can). If we need it again, let's move these utils in a core package.

https://algolia.atlassian.net/browse/FX-2503

### CODE REVIEW

File: packages/react-instantsearch-hooks-web/src/widgets/__tests__/Stats.test.tsx [hunk 1]
```diff
@@ -168,7 +168,7 @@ describe('Stats', () => {
     });
   });
 
-  test('renders with deprecated translations (with a deprecation warning)', async () => {
+  test.skip('renders with deprecated translations (with a deprecation warning)', async () => {
```
Comment (Haroenv) [inline packages/react-instantsearch-hooks-web/src/widgets/__tests__/Stats.test.tsx:171]: why not removed?

File: packages/react-instantsearch-hooks/src/index.ts [hunk 2]
```diff
@@ -35,4 +35,3 @@ export * from './connectors/useStats';
 export * from './connectors/useToggleRefinement';
 export * from './hooks/useConnector';
 export * from './hooks/useInstantSearch';
-export * from './lib/warn';
```
Comment (Haroenv) [inline packages/react-instantsearch-hooks/src/index.ts:38]: other usages of warn need to be removed too

File: packages/react-instantsearch-hooks-web/src/widgets/__tests__/Stats.test.tsx (hunk 1 above)
Comment (sarahdayan) [reply packages/react-instantsearch-hooks-web/src/widgets/__tests__/Stats.test.tsx:171]: Whoops, accidentally left it in.

File: packages/react-instantsearch-hooks/src/index.ts (hunk 2 above)
Comment (sarahdayan) [reply packages/react-instantsearch-hooks/src/index.ts:38]: Ah, I didn't notice it was used by other code sections after I exported it. Reintroducing it so it doesn't break the build, we'll move it later.

**Review by Haroenv — DISMISSED** [review]
use and the warnings on import also need to be removed in v7 :)

File: packages/react-instantsearch-hooks-web/src/widgets/__tests__/Stats.test.tsx (hunk 1 above)
Comment (Haroenv) [reply packages/react-instantsearch-hooks-web/src/widgets/__tests__/Stats.test.tsx]: it should be deleted, no?

**Review by dhayab — APPROVED** [review]

**Comment by dhayab** [comment]
We could wait for https://github.com/algolia/instantsearch/pull/5781 to be merged, to integrate the removal of `use`.

**Review by Haroenv — APPROVED** [review]
this is good, but use and the import warnings also need to be removed

**Comment by sarahdayan** [comment]
@Haroenv Waiting for #5781 to be merged to remove `use`.

**Review by dhayab — APPROVED** [review]

**Review by Haroenv — APPROVED** [review]

**Pull request merged**